Update hardening Kubernetes clusters URL in website to the latest version 1.2

seifrajhi opened the pull request:
NSA and CISA released the 1.0 version of the Kubernetes hardening guide in August 2021 and updated it based on industry feedback in March 2022 (version 1.1). The most recent version of the Kubernetes hardening guidance was released in August 2022 with corrections and clarifications.

<!--

 Hello!

 Remember to ADD A DESCRIPTION and delete this note before submitting
 your pull request. The description should explain what will change,
 and why.

 PLEASE title the FIRST commit appropriately, so that if you squash all
 your commits into one, the combined commit message makes sense.
 For overall help on editing and submitting pull requests, visit:
  https://kubernetes.io/docs/contribute/suggesting-improvements/

 Use the default base branch, “main”, if you're documenting existing
 features in the English localization.

 If you're working on a different localization (not English), see
 https://kubernetes.io/docs/contribute/new-content/overview/#choose-which-git-branch-to-use
 for advice.

 If you're documenting a feature that will be part of a future release, see
 https://kubernetes.io/docs/contribute/new-content/new-features/ for advice.

-->

sftim reviewed (changes requested):
Thanks for spotting that there's an update available.

**The change doesn't make sense**: the background to this article only includes things that had happened _prior_ to publication. It will jar with readers if the article appears to have been written taking into account guidance published a year later.

You could reframe the change as an aside to the article text. Please check in with the 3 article authors to make sure they're happy with the proposed change.

Ritikaa96 commented on content/en/blog/_posts/2021-10-05-nsa-cisa-hardening.md:
```diff
@@ -17,8 +17,7 @@ and are in no way a direct recommendation from the Kubernetes community or autho
 Security Agency (CISA)
-released, "[Kubernetes Hardening Guidance](https://media.defense.gov/2021/Aug/03/2002820425/-1/-1/1/CTR_KUBERNETES%20HARDENING%20GUIDANCE.PDF)"
-on August 3rd, 2021. The guidance details threats to Kubernetes environments
+released the 1.0 version of the Kubernetes hardening guide in August 2021 and updated it based on industry feedback in March 2022 (version 1.1). The most recent version of the Kubernetes hardening guidance was released in August 2022 with corrections and clarifications. Version 1.2 outlines a number of recommendations for  "[Kubernetes Hardening Guidance](https://media.defense.gov/2022/Aug/29/200 ...
```
```suggestion
released the 1.0 version of the Kubernetes hardening guide in August 2021 and updated it
based on industry feedback in March 2022 (version 1.1).The most recent version of the Kubernetes hardening guidance
was released in August 2022 with corrections and clarifications.
Version 1.2 outlines a number of recommendations for  "[Kubernetes Hardening Guidance](https://media.defense.gov/2022/Aug/29/2003066362/-1/-1/0/CTR_KUBERNETES_HARDENING_GUIDANCE_1.2_20220829.PDF)".
The guidance details threats to Kubernetes environments
```

seifrajhi commented:
> It will jar with readers if the article appears to have been written taking into account guidance published a year later.

The [current link](https://media.defense.gov/2021/Aug/03/2002820425/-1/-1/1/CTR_KUBERNETES%20HARDENING%20GUIDANCE.PDF) in this blog gives 404 error, so it should be changed or removed imo.

> You could reframe the change as an aside to the article text. 

I will do that 👍 ,  it's a better approach.


> Please check in with the 3 article authors to make sure they're happy with the proposed change.

I will try to contact them, to ask them if they are ok with  my changes.

PushkarJ commented:
This blog was actually one of the inputs taken as feedback to v1.0 and then the newer published version of hardening guide added the blog as a reference. 

Adding a line at the top as a markdown "note" mentioning that and then linking the v1.1 version as part of it would make more sense to avoid circular referencing :) 

Thank you @seifrajhi for spotting this and opening a PR to fix it.

seifrajhi commented:
Hey @PushkarJ ,

thank you for the prompt feedback  !!

That's exactly what I did, I added a note saying that there are newer versions v1.1 and v1.2 and I updated the PDF link to the latest version v1.2 ( v1.0 is not accessible anymore on the current link used in the blog )

sftim commented on content/en/blog/_posts/2021-10-05-nsa-cisa-hardening.md:
```diff
@@ -29,6 +29,11 @@ _Note_: This blog post is not a substitute for reading the guide. Reading the pu
 guidance is recommended before proceeding as the following content is 
 complementary.
 
+> The National Security Agency (NSA) and the Cybersecurity and Infrastructure Security Agency (CISA) released
+> the 1.0 version of the Kubernetes hardening guide in August 2021 and updated it based on industry feedback in March 2022 (version 1.1).
+> The most recent version of the Kubernetes hardening guidance was released in August 2022 with corrections and clarifications.
+> Version 1.2 outlines a number of recommendations for (hardening Kubernetes clusters)[https://media.defense.gov/2022/Aug/29/2003066362/-1/-1/0/CTR_KUBERNETES_HARDENING_GUIDANCE_1.2_20220829.PDF]. 
```
Thanks, but: please don't mark this as a quote (it's not a quote). Try using italics instead; for example:
```markdown
**Update, November 2023**  
_The National Security Agency (NSA) and the Cybersecurity and Infrastructure Security Agency (CISA)_
_released the 1.0 version of the Kubernetes hardening guide in August 2021 and updated it based on_
_industry feedback in March 2022 (version 1.1)._  
_The most recent version of the Kubernetes hardening guidance was released in August 2022_
_with corrections and clarifications._  
_Version 1.2 outlines a number of recommendations for_
[_hardening Kubernetes clusters_](https://media.defense.gov/2022/Aug/29/2003066362/-1/-1/0/CTR_KUBERNETES_HARDENING_GUIDANCE_1.2_20220829.PDF]. 
```

:information_source: The trailing whitespace is intentional and significant.

seifrajhi replied:
Changes are made per suggestion @sftim

seifrajhi replied:
Hello, any other changes to be made ?

PushkarJ commented:
I am okay with these new changes with a note. 

Tagging the remaining authors for review 

/cc @jimangel @savitharaghunathan

savitharaghunathan commented:
I am fine with the changes but will defer accepting this PR to sig-docs leads. IIRC, there used to be a policy on not updating blogposts. I am not sure if there are exceptions. Tagging co-chairs and apologies in advance for spamming.
cc: @divya-mohan0209 @reylejano @natalisucks

divya-mohan0209 commented:
@seifrajhi The update seems to have been formatted incorrectly in the [deploy preview](https://deploy-preview-44031--kubernetes-io-main-staging.netlify.app/blog/2021/10/05/nsa-cisa-kubernetes-hardening-guidance/). Could you please advise if this was intentional? If not, please could you fix it at the earliest possible?

sftim commented on content/en/blog/_posts/2021-10-05-nsa-cisa-hardening.md:
```diff
@@ -29,6 +29,18 @@ _Note_: This blog post is not a substitute for reading the guide. Reading the pu
 complementary.
 
+**Update, November 2023**  
+_The National Security Agency (NSA) and the Cybersecurity and Infrastructure Security Agency (CISA)_  
+_released the 1.0 version of the Kubernetes hardening guide in August 2021 and updated it based on_  
+_industry feedback in March 2022 (version 1.1)._  
+_The most recent version of the Kubernetes hardening guidance was released in August 2022_  
+_with corrections and clarifications._  
+_Version 1.2 outlines a number of recommendations for_  
+[_hardening Kubernetes clusters_](https://media.defense.gov/2022/Aug/29/2003066362/-1/-1/0/CTR_KUBERNETES_HARDENING_GUIDANCE_1.2_20220829.PDF).
```
Hmm, I didn't expect this to wrap like it has. Please amend so it renders as a paragraph.

seifrajhi replied:
ok I change it to paragraph without italic font

sftim commented:
Please put back the emphasis - even if that means making it all one line, or something. It's useful to highlight that this is an aside to the main article.

sftim commented:
You could use a [pageinfo](https://www.docsy.dev/docs/adding-content/shortcodes/#pageinfo) - does that work?

divya-mohan0209 commented:
Fwiw: The comment was not on the emphasis but the actual rendering of the text that was added. It wasn't wrapped correctly and my question to @seifrajhi was regarding that. I can see how that wasn't clear in my previous statement and I apologize.

seifrajhi commented:
@sftim, I used a `pageinfo` in the latest commit, it looks better in my opinion

waiting for your feedback

sftim commented:
/sig security
/label tide/merge-method-squash

sftim commented on content/en/blog/_posts/2021-10-05-nsa-cisa-hardening.md:
```diff
@@ -29,6 +29,14 @@ _Note_: This blog post is not a substitute for reading the guide. Reading the pu
+
+The National Security Agency (NSA) and the Cybersecurity and Infrastructure Security Agency (CISA) released the 1.0 version of the Kubernetes hardening guide in August 2021 and updated it based on industry feedback in March 2022 (version 1.1).
+
+The most recent version of the Kubernetes hardening guidance was released in August 2022 with corrections and clarifications. Version 1.2 outlines a number of recommendations for [hardening Kubernetes clusters](https://media.defense.gov/2022/Aug/29/2003066362/-1/-1/0/CTR_KUBERNETES_HARDENING_GUIDANCE_1.2_20220829.PDF).
```
Even better, IMO:
```suggestion
A more recent version of the Kubernetes hardening guidance was released in August 2022 with corrections and clarifications. Version 1.2 outlines a number of recommendations for [hardening Kubernetes clusters](https://media.defense.gov/2022/Aug/29/2003066362/-1/-1/0/CTR_KUBERNETES_HARDENING_GUIDANCE_1.2_20220829.PDF).
```

sftim reviewed (commented):
Thanks

/lgtm
/approve
